src/libraries/Common/src/Interop/Unix/System.Net.Security.Native/Interop.NetSecurityNative.cs

```diff
@@ -169,5 +172,17 @@ internal static partial class NetSecurityNative
 
             return Unwrap(out minorStatus, contextHandle, inputBytes, offset, count, ref outBuffer);
         }
+
+        // This constructor is added to address the issue with net6 regarding 
+        // Shim gss api on Linux to delay loading libgssapi_krb5.so
+        // issue https://github.com/dotnet/SqlClient/issues/1390
+        // dotnet runtime issue https://github.com/dotnet/runtime/pull/55037
+        static NetSecurityNative()
+        {
+            if (Environment.Version.Major >= 6)
```

Comment:
This file is only included in .NET 7+ binaries in this repo, so this check is unnecessary.

Reply:
And the EnsureGssInitialized calls are already being done in the main branch: https://github.com/dotnet/runtime/blob/3806e15595632e4bc8e4c4fb4cde12163006eca9/src/libraries/Common/src/Interop/Unix/System.Net.Security.Native/Interop.NetSecurityNative.IsNtlmInstalled.cs#L17-L20 . It is why you are getting the build breaks.

Perhaps you meant to create a PR against one of the servicing branches?